### PR TITLE
Add filters dependencies

### PR DESCRIPTION
**Pull request by mahdibenromdhane**
Added filter dependencies between: 
- Issuer => users, tags, company, sites, siteAreas.
- sites => siteAreas

### CODE REVIEW

File: src/app/pages/charging-stations/list/charging-stations-list-table-data-source.ts [hunk 1]
```diff
@@ -341,12 +341,13 @@ export class ChargingStationsListTableDataSource extends TableDataSource<Chargin
 
   public buildTableFiltersDef(): TableFilterDef[] {
     if (this.isOrganizationComponentActive) {
-      const siteFilter = new SiteTableFilter().getFilterDef();
+      const issuerFilter = new IssuerFilter().getFilterDef();
+      const siteFilter = new SiteTableFilter(null, [issuerFilter]).getFilterDef();
```
Comment (LucasBrazi06) [inline src/app/pages/charging-stations/list/charging-stations-list-table-data-source.ts:345]: SiteTableFilter should not get siteIDs anymore then only one param.

File: src/app/pages/organization/site-areas/list/site-areas-list-table-data-source.ts [hunk 2]
```diff
@@ -361,9 +361,10 @@ export class SiteAreasListTableDataSource extends TableDataSource<SiteArea> {
   }
 
   public buildTableFiltersDef(): TableFilterDef[] {
+    const issuerFilter = new IssuerFilter().getFilterDef();
     return [
-      new IssuerFilter().getFilterDef(),
-      new SiteTableFilter().getFilterDef(),
+      issuerFilter,
+      new SiteTableFilter(null, [issuerFilter]).getFilterDef(),
```
Comment (LucasBrazi06) [inline src/app/pages/organization/site-areas/list/site-areas-list-table-data-source.ts:367]: Same comment here, remove siteIDs param in constructor.

File: src/app/pages/transactions/history/transactions-history-table-data-source.ts [hunk 3]
```diff
@@ -371,12 +372,13 @@ export class TransactionsHistoryTableDataSource extends TableDataSource<Transact
...
-      filters.push(new SiteTableFilter().getFilterDef());
-      filters.push(new SiteAreaTableFilter().getFilterDef());
+      const siteFilter = new SiteTableFilter(null, [issuerFilter]).getFilterDef();
+      filters.push(siteFilter);
+      filters.push(new SiteAreaTableFilter([issuerFilter, siteFilter]).getFilterDef());
     }
     if (this.authorizationService.isAdmin() || this.authorizationService.hasSitesAdminRights()) {
-      filters.push(new UserTableFilter(this.authorizationService.getSitesAdmin()).getFilterDef());
-      filters.push(new TagTableFilter().getFilterDef());
+      filters.push(new UserTableFilter(this.authorizationService.getSitesAdmin(), [issuerFilter]).getFilterDef());
```
Comment (LucasBrazi06) [inline src/app/pages/transactions/history/transactions-history-table-data-source.ts:380]: To remove this.authorizationService.getSitesAdmin().
Sites are now handled in the backend.
Test the Site Admin role btw to be sure.

File: src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts [hunk 4]
```diff
@@ -276,17 +276,20 @@ export class TransactionsInProgressTableDataSource extends TableDataSource<Trans
 
   public buildTableFiltersDef(): TableFilterDef[] {
     const filters: TableFilterDef[] = [];
+    const issuerFilter = new IssuerFilter().getFilterDef();
     // Show Site Area Filter If Organization component is active
     if (this.componentService.isActive(TenantComponents.ORGANIZATION)) {
-      filters.push(new IssuerFilter().getFilterDef());
-      filters.push(new SiteTableFilter().getFilterDef());
-      filters.push(new SiteAreaTableFilter().getFilterDef());
+      const siteFilter = new SiteTableFilter(null, [issuerFilter]).getFilterDef();
```
Comment (LucasBrazi06) [inline src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts:282]: Same comments as above on siteIDs

File: src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts [hunk 5]
```diff
@@ -276,17 +276,20 @@ export class TransactionsInProgressTableDataSource extends TableDataSource<Trans
...
+      filters.push(siteFilter);
+      filters.push(new SiteAreaTableFilter([siteFilter, issuerFilter]).getFilterDef());
     }
     filters.push(new ChargingStationTableFilter().getFilterDef());
     filters.push(new ConnectorTableFilter().getFilterDef());
-    if (this.authorizationService.isAdmin() || this.authorizationService.hasSitesAdminRights()) {
-      filters.push(new UserTableFilter(this.authorizationService.getSitesAdmin()).getFilterDef());
-      filters.push(new TagTableFilter().getFilterDef());
+    if ((this.authorizationService.isAdmin() || this.authorizationService.hasSitesAdminRights())) {
+      filters.push(new UserTableFilter(this.authorizationService.getSitesAdmin(),
```
Comment (LucasBrazi06) [inline src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts:290]: Same comments as above on siteIDs

File: src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts [hunk 6]
```diff
@@ -276,17 +276,20 @@ export class TransactionsInProgressTableDataSource extends TableDataSource<Trans
...
+        this.componentService.isActive(TenantComponents.ORGANIZATION) ? [issuerFilter] : []).getFilterDef());
```
Comment (LucasBrazi06) [inline src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts:291]: issuerFilter has nothing to do with Organization check I think which can be removed.

File: src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts [hunk 7]
```diff
@@ -276,17 +276,20 @@ export class TransactionsInProgressTableDataSource extends TableDataSource<Trans
...
+      filters.push(new TagTableFilter(this.componentService.isActive(TenantComponents.ORGANIZATION) ? [issuerFilter] : []).getFilterDef());
```
Comment (LucasBrazi06) [inline src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts:292]: Same here about Organization

File: src/app/pages/users/list/users-list-table-data-source.ts [hunk 8]
```diff
@@ -393,7 +393,7 @@ export class UsersListTableDataSource extends TableDataSource<User> {
       new UserRoleFilter(this.centralServerService).getFilterDef(),
       new UserStatusFilter().getFilterDef(),
       new TagTableFilter([issuerFilter]).getFilterDef(),
-      new SiteTableFilter().getFilterDef(),
+      new SiteTableFilter(null, [issuerFilter]).getFilterDef(),
```
Comment (LucasBrazi06) [inline src/app/pages/users/list/users-list-table-data-source.ts:396]: Same comments as above on siteIDs

File: src/app/pages/users/tags/tags-list-table-data-source.ts [hunk 9]
```diff
@@ -382,10 +382,11 @@ export class TagsListTableDataSource extends TableDataSource<Tag> {
   }
 
   public buildTableFiltersDef(): TableFilterDef[] {
+    const issuerFilter = new IssuerFilter().getFilterDef();
     return [
-      new IssuerFilter().getFilterDef(),
+      issuerFilter,
       new StatusFilter().getFilterDef(),
-      new UserTableFilter().getFilterDef(),
+      new UserTableFilter(null, [issuerFilter]).getFilterDef(),
```
Comment (LucasBrazi06) [inline src/app/pages/users/tags/tags-list-table-data-source.ts:389]: Same comments as above on siteIDs

File: src/app/shared/table/filters/site-table-filter.ts [hunk 10]
```diff
@@ -3,7 +3,7 @@ import { FilterType, TableFilterDef } from '../../../types/Table';
 import { TableFilter } from './table-filter';
 
 export class SiteTableFilter extends TableFilter {
-  public constructor(siteIDs?: readonly string[]) {
+  public constructor(siteIDs?: readonly string[], dependentFilters?: TableFilterDef[]) {
```
Comment (LucasBrazi06) [inline src/app/shared/table/filters/site-table-filter.ts:6]: Same comments as above on siteIDs

File: src/app/shared/table/filters/user-table-filter.ts [hunk 11]
```diff
@@ -3,7 +3,7 @@ import { UsersDialogComponent } from '../../dialogs/users/users-dialog.component
 import { TableFilter } from './table-filter';
 
 export class UserTableFilter extends TableFilter {
-  public constructor(siteIDs?: readonly string[]) {
+  public constructor(siteIDs?: readonly string[], dependentFilters?: TableFilterDef[]) {
```
Comment (LucasBrazi06) [inline src/app/shared/table/filters/user-table-filter.ts:6]: Same comments as above on siteIDs, to remove

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/app/pages/transactions/history/transactions-history-table-data-source.ts (hunk 3 above)
Comment (mahdibenromdhane) [reply src/app/pages/transactions/history/transactions-history-table-data-source.ts]: Tested: the authorisations is handled in the backend

File: src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts (hunk 6 above)
Comment (mahdibenromdhane) [reply src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts]: Done

**Review by LucasBrazi06 — APPROVED** [review]

**Pull request merged**